pkg/connector/service_accounts.go

```diff
@@ -44,27 +47,40 @@ func serviceAccountResource(member string) (*v2.Resource, error) {
 // List returns all the users from the database as resource objects.
 // Users include a UserTrait because they are the 'shape' of a standard user.
 func (o *serviceAccountBuilder) List(ctx context.Context, parentResourceID *v2.ResourceId, pToken *pagination.Token) ([]*v2.Resource, string, annotations.Annotations, error) {
+	var resources []*v2.Resource
+	l := ctxzap.Extract(ctx)
+	projectId := o.BigQueryClient.Project()
+	if isExcluded(o.excludeProjectIDs, projectId) {
+		l.Warn(
+			"baton-google-bigquery: project in exclusion list",
+			zap.String("projectId", projectId),
+		)
+
+		return resources, "", nil, nil
+	}
+
 	policy, err := o.ProjectsClient.GetIamPolicy(ctx, &iampb.GetIamPolicyRequest{
 		Resource: fmt.Sprintf("projects/%s", o.BigQueryClient.Project()),
 	})
-	if err != nil {
-		return nil, "", nil, wrapError(err, "failed to get IAM policy")
+	if !isPermissionDenied(ctx, err) {
+		return nil, "", nil, wrapError(err, "listing service accounts failed")
 	}
 
-	var resources []*v2.Resource
-	for _, binding := range policy.Bindings {
-		for _, member := range binding.Members {
-			isServiceAccount, member := isServiceAccount(member)
-			if !isServiceAccount {
-				continue
-			}
+	if policy != nil {
```

Comment:
Early returning if policy == nil is simpler.

Reply:
Fixed